fv3core/stencils/saturation_adjustment.py

```diff
@@ -50,45 +50,31 @@ def tem_upper(i):
     return 253.16 + DELT * i
 
 
-# TODO this abstraction is not possible in gt4py as these are called in
-# conditionals and temporaries get created. but would be nice to have 1 version
-# of the equation.
-# {
-# @gtscript.function
-# def q_table_oneline(delta_heat_capacity, latent_heat_coefficient, tem):
-#     return E00 * exp((delta_heat_capacity * log(tem / TICE)
-#         + (tem - TICE) / (tem * TICE) *  latent_heat_coefficient) / constants.RVGAS)
-
-# @gtscript.function
-# def table_vapor_oneline(tem):
-#     return q_table_oneline(DC_VAP, LV0, tem)
-
-
-# @gtscript.function
-# def table_ice_oneline(tem):
-#     return q_table_oneline(D2ICE, LI2, tem)
-# }
+@gtscript.function
+def q_table_oneline(delta_heat_capacity, latent_heat_coefficient, tem):
+    return E00 * exp(
+        (
+            delta_heat_capacity * log(tem / TICE)
+            + (tem - TICE) / (tem * TICE) * latent_heat_coefficient
+        )
+        / constants.RVGAS
+    )
 
 
 @gtscript.function
 def table_vapor_oneline(tem):
-    return E00 * exp(
-        (DC_VAP * log(tem / TICE) + (tem - TICE) / (tem * TICE) * LV0) / constants.RVGAS
-    )
+    return q_table_oneline(DC_VAP, LV0, tem)
 
 
 @gtscript.function
 def table_ice_oneline(tem):
-    return E00 * exp(
-        (D2ICE * log(tem / TICE) + (tem - TICE) / (tem * TICE) * LI2) / constants.RVGAS
-    )
+    return q_table_oneline(D2ICE, LI2, tem)
 
 
 # TODO Math can be consolidated if we can call gtscript functions from
 # conditionals, fac0 and fac2 functions and others.
 @gtscript.function
 def qs_table_fn(i):
-    table = 0.0
```

Comment:
good catch, we no longer need to 'declare' the temporary variables before conditionals to ensure they exist and have a value regardless of the branching conditions? Since temporaries are sometimes just local scalars, I think this works, pretty helpful. Is it ok to do this? The coder just needs to make sure if they use the temporary after the condition, it got assigned to something in one of the branches

Reply:
I think this is fine since the value of `table` will be assigned at least in one of the first two conditions.  A better fix that gets rid of a conditional is to initially declare `table = 0.0` and get rid of the first `if i >= 1600...` conditional since it wouldn't be needed.

Reply:
either way is good by me! I'm so happy gt4py allows you to choose now